src/sentry/receivers/outbox/region.py

```diff
@@ -46,6 +46,21 @@ def process_user_ip_event(payload: Any, **kwds: Any):
         DatabaseBackedLogService().record_user_ip(event=UserIpEvent(**payload))
 
 
+@receiver(process_region_outbox, sender=OutboxCategory.ORGANIZATION_MEMBER_CREATE)
+def process_organization_member_create(
+    object_identifier: int, payload: Any, shard_identifier: int, **kwds: Any
+):
+    if (org_member := maybe_process_tombstone(OrganizationMember, object_identifier)) is None:
+        # Delete all identities that may have been associated.  This is an implicit cascade.
+        if payload and "user_id" in payload:
+            identity_service.delete_identities(
+                user_id=payload["user_id"], organization_id=shard_identifier
+            )
+        return
```

Comment:
Why does creating a member result in identities being removed?

Reply:
@markstory My understanding of why `maybe_process_tombstone()` is called here, and in other receivers, was that the object instance identified by `object_identifier` could be deleted by the time the receiver is called. 

Is that inaccurate?

Reply:
That is correct!  However, in that case, as written, an `ORGANIZATION_MEMBER_UPDATE` outbox will be generated, and it can deal with the delete.  In which case you wouldn't need this logic here, per say.  Thanks to ordering, the create will arrive before the delete in all cases.

Reply:
> Thanks to ordering, the create will arrive before the delete in all cases.

@corps where do we enforce this ordering? It's non-obvious to me 😬